int32 agent_id
task[] goal_list